src/algebra/star/module.lean

```diff
@@ -33,3 +33,17 @@ def star_linear_equiv (R : Type*) {A : Type*}
 { to_fun := star,
   map_smul' := star_smul,
   .. star_add_equiv }
+
+variables (R : Type*) {A : Type*}
+  [semiring R] [invertible (2 : R)] [star_monoid R] [has_trivial_star R]
+  [add_comm_group A] [module R A] [star_add_monoid A] [star_module R A]
+
+/-- The real part of an element of a star module, as a real-linear map. -/
+@[simps] def star_module.re : A →ₗ[R] self_adjoint A :=
```

Comment:
I'm not convinced about the name `re` here.

More importantly, I think that if we're going to generalise to any star module like this then we should be symmetric and do it for both the "real" (self-adjoint) and "imaginary" (skew-adjoint) parts.

If we have a involutive endomorphism `T` on some `A` then we have a direct sum decomposition of `A` into the two eigenspaces of `T`: the self-adjoint and skew-adjoint elements. For any `x : A`:
```
  x = (x + Tx)/2 + (x - Tx)/2
```

And then I think we should prove that in the case that the scalars are `ℂ` with complex conjugation, the scalar action of `I` gives an `ℝ`-linear equivalence between `skew_adjoint A` and `self_adjoint A`.

Reply:
Does `skew_adjoint` already exist?

Reply:
Right, I guess now is as good a time as any to add `skew_adjoint`.

Reply:
This is now done. I renamed `re` to `self_adjoint_part`, added `skew_adjoint_part` and the decomposition in terms of those two, and added the original lemma as a specialization of this for the complex case.